app/models/topic_tracking_state.rb

```diff
@@ -577,6 +577,34 @@ def self.trigger_post_read_count_update(post, groups, last_read_post_number, use
     post.publish_change_to_clients!(:read, opts)
   end
 
+  def self.report_totals(user)
+    new, unread = 0, 0
+
+    report(user).each do |entry|
+      # any changes here should also be reflected in model/topic-tracking-state.js
+      if entry.last_read_post_number == nil &&
+           (
+             entry.notification_level == nil ||
+               entry.notification_level >= NotificationLevels.all[:tracking] &&
+                 entry.created_at > user.user_option.treat_as_new_topic_start_date
+           )
+        new += 1
+      end
+
+      if entry.last_read_post_number != nil &&
+           entry.last_read_post_number < entry.highest_post_number &&
+           entry.notification_level >= NotificationLevels.all[:tracking]
+        unread += 1
+      end
```

Comment:
Since this is a dedicated route, I think we can just do a count on the `self.report_raw_sql` query which is able to fetch the unread and new counts. It will be two queries but I would just let the database do the counting here.

Reply:
Yeah, that makes sense. Added it.